src/execution/physical_plan/plan_create_index.cpp

```diff
@@ -12,9 +14,27 @@ namespace duckdb {
 
 unique_ptr<PhysicalOperator> PhysicalPlanGenerator::CreatePlan(LogicalCreateIndex &op) {
 
+	// generate a physical plan for the parallel index creation which consists of the following operators
+	// table scan - projection (for expression execution) - filter (NOT NULL) - order - create index
+
 	D_ASSERT(op.children.empty());
 
+	// validate that all expressions contain valid scalar functions
+	// e.g. get_current_timestamp(), random(), and sequence values are not allowed as ART keys
+	// because they make deletions and lookups unfeasible
+	for (idx_t i = 0; i < op.unbound_expressions.size(); i++) {
+		auto &expr = op.unbound_expressions[i];
+		if (expr->expression_class == ExpressionClass::BOUND_FUNCTION) {
+			auto &func_expr = (BoundFunctionExpression &)*expr;
+			if (func_expr.function.name == "get_current_timestamp" || func_expr.function.name == "random" ||
```

Comment:
I wonder if this is too error-prone, since other extensions could also make functions, or supply a UDF
maybe BoundFunctionExpression needs a `IsDeterministic` flag of some similar?

Or maybe for now add a limit that the function has to come from the system catalog?

Reply:
Turns out that someone already thought of this! :) 
```c++
enum class FunctionSideEffects : uint8_t { NO_SIDE_EFFECTS = 0, HAS_SIDE_EFFECTS = 1 };
...
//! Whether or not the function has side effects (e.g. sequence increments, random() functions, NOW()). Functions
//! with side-effects cannot be constant-folded.
FunctionSideEffects side_effects;
```